android/codepush.gradle

```diff
@@ -23,8 +23,13 @@ android.buildTypes.each { buildType ->
     buildType.resValue 'string', "CODE_PUSH_APK_BUILD_TIME", String.format("\"%d\"", System.currentTimeMillis())
 }
 
-gradle.projectsEvaluated { 
+gradle.projectsEvaluated {
     android.applicationVariants.all { variant ->
+        // skip debug build variant
+        if (variant.buildType.name == "debug") {
```

Comment:
Not sure if this is correct or if this is where it's supposed to fall into the other side of the if

Reply:
@johnf , Does this condition fix something? Do we need it?

Reply:
So the new React Native Gradle Plugin doesn't create a bundle task for debug builds.
So this was failing for me without it.
But this was the first change I made before I fixed the production builds.
It's possible that the else that's complimentary to this code is what's supposed to run in dev.
I'll see if I can do some more testing tomorrow and work out if this stays or now._

Reply:
Got it. What about the back compatibility? Should we fallback to old code for older gradle plugin?

Reply:
@DmitriyKirakosyan I'm about to push some change

Re the debug skip - the new plugin skips creating a bundle in debuggable builds entirely. Which makes sense. I think normally, you would be connecting to metro. I have used a similar code to the new plugin to skip code push in the same way.

The one thing I'm confused about is this code path https://github.com/microsoft/react-native-code-push/blob/master/android/codepush.gradle#L69-L108

For it to execute, ```bundleJsAndAssets``` has not to exist, but then in the first line after this else block, we depend on ```bundle${target}JsAndAssets```. So I can't understand how it could have ever worked in its current form. As far as I know, rails would always have included this task.

Is it historical and just hasn't been relevant for a while? I would be tempted to remove it entirely, but I'm not familiar enough to know if that's a good idea.

Reply:
> Got it. What about the back compatibility? Should we fallback to old code for older gradle plugin?

If you look at the new changes (I discovered that the react config wasn't being pulled in) they are pretty substantial. I think maintaining backward compatibility would be difficult.

Given the 0.71 upgraders suggest switching everything across to the new plugin, I think we avoid backward compatibility and update the table in the Readme (which I've done)

Reply:
I've seen other libs drop support for version below 0.71 due to the complexity of maintaining both so it seems reasonable to me

Reply:
Am I understanding this right that v8.0.0+ will only support RN v0.71+ on Android? Right now the compatibility table in the README doesn't seem to indicate that if so.